lib/xeroizer/generic_application.rb

```diff
@@ -63,7 +63,7 @@ class GenericApplication
       # @see PublicApplication
       # @see PrivateApplication
       # @see PartnerApplication
-      def initialize(consumer_key, consumer_secret, options = {})
+      def initialize(client, options = {})
```

Comment:
With this, may be worth doing a major version bump, so people are aware of the constructor change.

Reply:
The consumers likely shouldn't see any change, since we didn't change the other clients.  However, if they depend on the generic application then there would be a problem.  

We've strived to maintain passing tests in the unit and integration/acceptance suite.

Reply:
@CloCkWeRX I'll let you make the determination as to whether or not you want to make a major version bump here.

Reply:
I'm happy to skip it then